Simplenote/src/androidTest/java/com/automattic/simplenote/integration/TagsActivityTest.kt

```diff
@@ -0,0 +1,248 @@
+package com.automattic.simplenote.integration
+
+import android.widget.EditText
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.RecyclerViewActions.scrollToPosition
+import androidx.test.espresso.matcher.RootMatchers
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.filters.MediumTest
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner
+import com.automattic.simplenote.BaseUITest
+import com.automattic.simplenote.R
+import com.automattic.simplenote.TagsActivity
+import com.automattic.simplenote.utils.isToast
+import com.automattic.simplenote.utils.withRecyclerView
+import org.hamcrest.CoreMatchers
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4ClassRunner::class)
+@MediumTest
+class TagsActivityTest : BaseUITest() {
+
+    @Test
+    fun listTagsShouldShowCompleteListTags() {
+        createTag("tag1")
+        createTag("tag2")
+        createTag("tag3")
+        createTag("other")
+        // To edit tags, tags should belong a note
+        createNote("Hello1", listOf("tag1", "tag2"))
+        createNote("Hello2", listOf("tag2"))
+        createNote("Hello3", listOf("tag1", "tag2", "tag3"))
+
+        assertEquals(tagsBucket.count(), 4)
```

Comment:
I like `AssertJ`. I think fluent assertions are easier to read but since the assertions that were not related to Espresso were simple enough I decided to use JUnit assertions. I think we can add `AssertJ` if we add more complex assertions.

Reply:
Makes sense, thanks! We can use the JUnit assertions for now 👍

Reply:
I agree. I have been using [Strikt](https://strikt.io/) for some time now, which is kind of `AssertJ` for Kotlin, and I like it. 😄